### PR TITLE
USWDS - Banner: remove layout grid dependency

### PR DESCRIPTION
**Pull request by mejiaj**
## Description

**Remove layout dependency.** Removes the SASS layout grid dependency from component markup.

### Preview
- [Banner default](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/jm-usa-banner-package/?path=/story/components-banner--default)
- [Default Spanish](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/jm-usa-banner-package/?path=/story/components-banner--default-spanish)
- [Mil](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/jm-usa-banner-package/?path=/story/components-banner--mil)
- [Mil Spanish](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/jm-usa-banner-package/?path=/story/components-banner--mil-spanish)

### Major changes

- Update test template markup 98b0839bf944c21fde37920f99d72d5610921e9d
- CSS selectors match order of markup 551972b2ec31816d53036ddae1b692f6c7dd692e
- Removed layout grid classes bfd2f8907d277dd5063b97de98de94b6c6401caf

### Compiled CSS before & after

| Name       | Full size     | Minified size  |
| :---------- | ------------- | -------------- |
| Banner old | 48KB          | 38KB           |
| Banner new | 11KB  (-37KB) | 9KB    (-29KB) |

Before you hit Submit, make sure you’ve done whichever of these applies to you:

- [ ] Follow the [18F Front End Coding Style Guide](https://pages.18f.gov/frontend/) and [Accessibility Guide](https://pages.18f.gov/accessibility/checklist/).
- [x] Run `npm test` and make sure the tests for the files you have changed have passed.
- [ ] Run your code through [HTML_CodeSniffer](http://squizlabs.github.io/HTML_CodeSniffer/) and make sure it’s error free.
- [x] Title your pull request using this format: [Website] - [UI component]: Brief statement describing what this pull request solves.

### CODE REVIEW

File: packages/usa-banner/src/usa-banner.twig [hunk 1]
```diff
@@ -12,32 +12,49 @@
   <div class="usa-accordion">
     <header class="usa-banner__header">
       <div class="usa-banner__inner">
-        <div class="grid-col-auto">
-          <img class="usa-banner__header-flag" src="./img/us_flag_small.png" alt="U.S. flag">
-        </div>
-        <div class="grid-col-fill tablet:grid-col-auto">
+        <img class="usa-banner__header-flag" src="./img/us_flag_small.png" alt="U.S. flag">
+        <div class="usa-banner__header-content">
```
Comment (mejiaj) [inline packages/usa-banner/src/usa-banner.twig:16]: Had to add in a container to group all text content and expand button.  

Equivalent to the previous div on line 18 `<div class="grid-col-fill tablet:grid-col-auto">`

File: packages/usa-banner/src/styles/_usa-banner.scss [hunk 2]
```diff
@@ -143,6 +112,7 @@ $banner-icon-close: (
 }
 
 .usa-banner__header-action {
+  @include u-margin-y(0);
```
Comment (amyleadem) [inline packages/usa-banner/src/styles/_usa-banner.scss:115]: This narrows the space between paragraphs by setting margin-top:0. Prior version had margin-top: 2px.

File: packages/usa-banner/src/usa-banner.twig [hunk 3]
```diff
@@ -12,32 +12,49 @@
...
           <p class="usa-banner__header-text">{{ banner.text }}</p>
-          <p class="usa-banner__header-action" aria-hidden="true">{{ banner.action }}</p>
+          <p class="usa-banner__header-action" aria-hidden="true">
+            {{ banner.action }}
```
Comment (amyleadem) [inline packages/usa-banner/src/usa-banner.twig:19]: There appears to be an extra space to the end of the action text that isn't there in the prior version. 
image

File: packages/usa-banner/src/test/template.html [hunk 4]
```diff
@@ -9,38 +9,43 @@
...
-          <span class="usa-banner__button-text">Here's how you know</span>
+        <button class="usa-accordion__button usa-banner__button" aria-expanded="false"
+          aria-controls="gov-banner-default">
+          <span class="usa-banner__button-text">Here’s how you know</span>
         </button>
       </div>
     </header>
-    <div class="usa-banner__content usa-accordion__content" id="gov-banner">
+    <div class="usa-banner__content usa-accordion__content" id="gov-banner-default" hidden="">
       <div class="grid-row grid-gap-lg">
```
Comment (amyleadem) [inline packages/usa-banner/src/test/template.html:19]: I see grid references here that were removed in usa-banner.twig. Should they be removed in this file as well?

File: packages/usa-banner/src/test/template.html [hunk 5]
```diff
@@ -9,38 +9,43 @@
...
-        >
-          <span class="usa-banner__button-text">Here's how you know</span>
+        <button class="usa-accordion__button usa-banner__button" aria-expanded="false"
+          aria-controls="gov-banner-default">
+          <span class="usa-banner__button-text">Here’s how you know</span>
         </button>
       </div>
     </header>
-    <div class="usa-banner__content usa-accordion__content" id="gov-banner">
+    <div class="usa-banner__content usa-accordion__content" id="gov-banner-default" hidden="">
```
Comment (amyleadem) [inline packages/usa-banner/src/test/template.html:18]: Curious - why was hidden added here?

**Comment by amyleadem** [comment]
Looks good! Way to shrink the package down. 

I completed the following checks:
- [x] Compared appearance to current state at multiple browser widths in Firefox, Chrome, Safari, and Edge for MacOS
- [x] Reviewed html and scss updates

Found a few small differences and added comments in the related file. Let me know if you have questions.

File: packages/usa-banner/src/styles/_usa-banner.scss (hunk 2 above)
Comment (mejiaj) [reply packages/usa-banner/src/styles/_usa-banner.scss]: Yes, it's added to the action on mobile **only**. It seemed small enough to remove, but I've added back in [86aa75ca941cd168ce3ffeb73555fed4428773e8].

image

File: packages/usa-banner/src/usa-banner.twig (hunk 3 above)
Comment (mejiaj) [reply packages/usa-banner/src/usa-banner.twig:19]: Funnily enough, this is caused from breaking the markup into its own line.

```diff
- <p class="usa-banner__header-action" aria-hidden="true">{{ banner.action }}</p>
+ <p class="usa-banner__header-action" aria-hidden="true">
+  {{ banner.action }}
+ </p>
```

Fixed in 4910eb324738f3d1404b79687e5b1d947f38ce80.

File: packages/usa-banner/src/test/template.html (hunk 4 above)
Comment (mejiaj) [reply packages/usa-banner/src/test/template.html]: Totally, removed in b50e5a6eb3d24893d13310e7290b42d8da683e18. Good catch!

File: packages/usa-banner/src/test/template.html (hunk 5 above)
Comment (mejiaj) [reply packages/usa-banner/src/test/template.html]: Me too, not sure where this came from. It shouldn't be there and removed as part of b50e5a6.

**Comment by mejiaj** [comment]
@amyleadem thanks for the great feedback, I've addressed comments above. Please check them out when you have a chance!

**Review by amyleadem — APPROVED** [review]
Great work! Everything looks good to me

**Pull request closed without merging**